lib/v2/screens/dashboard/wallet_screen.dart

```diff
@@ -23,37 +27,50 @@ class WalletScreen extends StatefulWidget {
 }
 
 class _WalletScreenState extends State<WalletScreen> with AutomaticKeepAliveClientMixin {
-
   @override
   bool get wantKeepAlive => true;
 
   @override
   Widget build(BuildContext context) {
     super.build(context);
     return BlocProvider(
-      create: (_) => WalletBloc()..add(const RefreshDataEvent()),
+      create: (_) => WalletBloc()..add(LoadUserValues()),
       child: BlocBuilder<WalletBloc, WalletState>(
         builder: (context, state) {
-          return RefreshIndicator(
-            onRefresh: () async {
-              BlocProvider.of<RatesBloc>(context).add(const FetchRates());
-              BlocProvider.of<WalletBloc>(context).add(const RefreshDataEvent());
-            },
-            child: Scaffold(
-              appBar: buildAppBar(context) as PreferredSizeWidget?,
-              body: ListView(
-                // TODO(n13): Use exact measurements from figma
-                children: <Widget>[
-                  const SizedBox(height: 15),
-                  const TokenCards(),
-                  const SizedBox(height: 20),
-                  buildSendReceiveButton(context),
-                  const SizedBox(height: 20),
-                  walletBottom(context),
-                ],
-              ),
-            ),
-          );
+          switch (state.pageState) {
+            case PageState.loading:
+              return const FullPageLoadingIndicator();
+            case PageState.failure:
+              return const FullPageErrorIndicator();
+            case PageState.success:
+              return RefreshIndicator(
+                onRefresh: () async {
+                  BlocProvider.of<RatesBloc>(context).add(const FetchRates());
+                  BlocProvider.of<WalletBloc>(context).add(const RefreshDataEvent());
+                },
+                child: Scaffold(
+                  appBar: buildAppBar(
+                    context,
+                    state.profile!.account,
+                    state.profile!.nickname,
+                    state.profile!.image,
+                  ) as PreferredSizeWidget,
```

Comment:
No widget class? WalletAppBar??

Reply:
I try to add one but keep getting a preferredSize error. and I dont want to give the appbar a size.